docs/DeveloperGuide.md

```diff
@@ -1,22 +1,82 @@
 # Developer Guide
 
 ## Acknowledgements
-[Addressbook-level3](https://github.com/se-edu/addressbook-level3)
+- **round() method in Cashflow.java**
+  - author: mhadidg
+  - source: https://stackoverflow.com/questions/2808535/round-a-double-to-2-decimal-places
+- **capitalize() method in Cashflow.java**
+  - author: Nick Bolton
+  - source: https://stackoverflow.com/questions/1892765/how-to-capitalize-the-first-character-of-each-word-in-a-string
+    
+- **DG adapted from**
+  - [Addressbook-level3](https://github.com/se-edu/addressbook-level3)
+  
+  
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+
+## Components
+### Storage Component
+API: `Storage.java`
+
+![](images/Storage.png)
+
+- The storage component loads data from the saved text files when the application starts, and saves the data to the
+  text files when the application exits.
+- The storage class uses the static methods in LoadData and SaveData to load and save data respectively.
+- The `load` method in LoadData reads the `data.txt` file and loads any existing Income, Expense and Budget into the application.
+- The `save` method in SaveData saves all Incomes, Expenses and existing Budget into the `data.txt` file.
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
-### Storage Component
-API: `Storage.java`
 
-![](images/Storage.png)
+### Add income/expense feature
+
+The add income/expense command has 2 compulsory arguments `/t` and `/a` and 1 optional argument `/r`.
+
+Example:
+```
+add income /a 100 /t salary /r 30
+```
+Below are the steps that shows the implementation of add income/expense.
+#### Step 1
+An instantiated AddCashflowCommand class gets the instance of CashflowList.
+
+This allows the AddCashflowCommand instance to access the methods of CashflowList.
+#### Step 2
+The AddCashflowCommand instance then calls addIncome() or addExpense(), depending on what `category` is initialised as.
+
+addIncome() or addExpense() instantiates an Income or Expense object respectively.
+
+Example:
+```
+switch (category) {
+        case INCOME:
+            cashflowList.addIncome(amount, incomeType, recur);
+            break;
+        case EXPENSE:
+            cashflowList.addExpense(amount, expenseType, recur);
+            break;
+        default:
+            ui.showMessage("Unidentified entry.");
+            break;
+        }
+```
```

Comment:
Maybe can include a screenshot of an example output here!

Reply:
very good suggestion, will consider